sui_programmability/examples/nfts/sources/collectible.move

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Small and simple implementation for the common collectible type.
+/// Contains a basic set of fields, the only required one of which is `img_url`.
+///
+/// Other fields can be omitted by using an `option::none()`.
+/// Custom metadata can be created and passed into the `Collectible` but that would
+/// require additional work on the creator side to set up metadata creation methods.
+module nfts::collectible {
+    use sui::object::{Self, UID};
+    use sui::publisher::{Self, Publisher};
+    use sui::tx_context::{TxContext};
+    use std::option::{Self, Option};
+    use std::string::String;
+    use std::vector as vec;
+
+    /// For when a witness type passed is not an OTW.
+    const ENotOneTimeWitness: u64 = 0;
+    /// For when the type `T` is not from the same module as the OTW.
+    const EModuleDoesNotContainT: u64 = 1;
+    /// For when maximum size of the Collection is reached - minting forbidden.
+    const ECapReached: u64 = 2;
+    /// For when Names length does not match `img_urls` length
+    const EWrongNamesLength: u64 = 3;
+    /// For when Descriptions length does not match `img_urls` length
+    const EWrongDescriptionsLength: u64 = 4;
+    /// For when Creators length does not match `img_urls` length
+    const EWrongCreatorsLength: u64 = 5;
+    /// For when Metadatas length does not match `img_urls` length
+    const EWrongMetadatasLength: u64 = 6;
+
+    /// Basic collectible - should contain only unique information (eg
+    /// if all collectibles have the same description, it should be put
+    /// into the Display to apply to all of the objects of this type, and
+    /// not in every object).
+    struct Collectible<T: store> has key, store {
+        id: UID,
+        /// The only required parameter for the Collectible.
+        /// Should only contain a unique part of the URL to be used in the
+        /// template engine in the `Display` and save gas and storage costs.
+        img_url: String,
+        name: Option<String>,
```

Comment:
The options on name/description/creator seem a bit cumbersome--maybe we can get rid of them and folks can supply the empty string if they don't want these?

Reply:
I wouldn't. It's going to be really hard to detect an empty string in Display.